core/ipfs.go

```diff
@@ -0,0 +1,72 @@
+package core
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"gx/ipfs/QmXYjuNuxVzXKJCfWasQk1RqkhVLDM9jtUKhqc2WPQmFSB/go-libp2p-peer"
+	"io/ioutil"
+	"os"
+	"path"
+	"time"
+
+	"github.com/OpenBazaar/openbazaar-go/ipfs"
+	"github.com/ipfs/go-ipfs/namesys"
+	ipfsPath "github.com/ipfs/go-ipfs/path"
+)
+
+var getModelFromIPFSCache = map[string]getModelFromIPFSCacheEntry{}
+
+type getModelFromIPFSCacheEntry struct {
+	bytes   []byte
+	created time.Time
+}
+
+// PublishModelToIPFS writes the given data to IPFS labeled as the given model
+// name
+func (n *OpenBazaarNode) PublishModelToIPFS(model string, data interface{}) error {
+	serializedData, err := json.MarshalIndent(data, "", "    ")
+	if err != nil {
+		return err
+	}
+	h := sha256.Sum256(serializedData)
+
+	tmpPath := path.Join(n.RepoPath, hex.EncodeToString(h[:])+".json")
+	err = ioutil.WriteFile(tmpPath, serializedData, os.ModePerm)
+	if err != nil {
+		return err
+	}
+	hash, err := ipfs.AddFile(n.Context, tmpPath)
+	if err != nil {
+		return err
+	}
+	err = os.Remove(tmpPath)
+	if err != nil {
+		return err
+	}
+
+	return ipfs.PublishAltRoot(n.Context, n.IpfsNode, model, ipfsPath.FromString("/ipfs/"+hash), time.Now().Add(namesys.DefaultPublishLifetime))
+}
+
+// GetModelFromIPFS gets the requested model from ipfs or the local cache
+func (n *OpenBazaarNode) GetModelFromIPFS(p peer.ID, model string, maxCacheLen time.Duration) ([]byte, error) {
+	entry, ok := getModelFromIPFSCache[p.Pretty()+"|"+model]
```

Comment:
not seeing where entries are actually saved to this cache

Reply:
🤦‍♂️ fixed